### PR TITLE
Containerapp 0.3.10 Release

### PR DESCRIPTION
**Pull request by StrawnSC**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Containerapp

**Comment by panchagnula** [comment]
@StrawnSC / @haroonf when is this version going to be ready for release? Seems like it missing some other changes we discussed should be part of the release?

**Comment by StrawnSC** [comment]
> @StrawnSC / @haroonf when is this version going to be ready for release? Seems like it missing some other changes we discussed should be part of the release?

@panchagnula is this the change you were talking about: https://github.com/calvinsID/azure-cli-extensions/pull/145 (Rune's fix for the `--location` in `az containerapp hostname list`)? 
The only other outstanding change is the "no dockerfile" feature, but I thought we wanted to give some time for @anthonychu to try it out. I wouldn't be opposed to merging that with this release though

**Comment by panchagnula** [comment]
> > @StrawnSC / @haroonf when is this version going to be ready for release? Seems like it missing some other changes we discussed should be part of the release?
> 
> @panchagnula is this the change you were talking about: [calvinsID#145](https://github.com/calvinsID/azure-cli-extensions/pull/145) (Rune's fix for the `--location` in `az containerapp hostname list`)? The only other outstanding change is the "no dockerfile" feature, but I thought we wanted to give some time for @anthonychu to try it out. I wouldn't be opposed to merging that with this release though

if we need to wait on testing/ sign-off from Anthony that is ok - but we need to get the regression fix deployed soon, so lets prioritize that accordingly. Thanks!

**Comment by panchagnula** [comment]
@zhoxing-ms could you review this today & share your initial comments!

**Review by zhoxing-ms — APPROVED** [review]

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py
Comment (panchagnula) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:568]: In general can we work with core-cli to figure out getting these tests run without live mode. Given extension tests don't run in nightly or release builds, or PR builds, these don't help until a dev runs all tests locally.
The containerapp-compose is supposed to have running these kind of tests, using a monkey patch of some sort, can we do something similar?

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py
Comment (panchagnula) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:586]: Curious instead this sort of validation, why can't we do a GET / show command & verify the app exists & some property for FDQN/ ingress or is as expected

**Review by panchagnula — CHANGES_REQUESTED** [review]
Please update the validation logic of the test. Create a work item at the min to investigate & work with core-cli how to make the "live only" tests (espcially for not "up") work as true scenario tests.

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py
Comment (smurawski) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:568]: The problem we bumped into was the generated name for the log analytics workspace.

We worked around this by adding [a hidden/suppressed parameter](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp-compose/azext_containerapp_compose/_params.py#L24) and then using the `self.create_random_name` to pass in a known value (known to the test harness) for the name of the log analytics workspace.

This required a monkey patch of `_generate_log_analytics_workspace_name` by https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp-compose/azext_containerapp_compose/_monkey_patch.py#L57-L60

I'd be happy to grab some time to pair to see if that'd be helpful for your tests.

Comment (panchagnula) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:568]: Thanks @smurawski - given this a create command that can take an existing log-analytics as an input I am curious why this failing. but to your point yes, happy to engage with you as needed.

Comment (smurawski) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:568]: @panchagnula - in this test (see line 575), it's doing a `containerapp env create` which generates a log analytics workspace with a random name.  This will fail on any attempt to replay the recorded test.

Comment (StrawnSC) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:559]: Thanks for the suggestion @smurawski! I'm testing this right now. Thankfully, we can pass in log analytics workspace names to `az containerapp env create` without needing to add hidden parameters to any of our commands, so this might be an easy fix for us

**Comment by StrawnSC** [comment]
@panchagnula I made the change to testing validation you mentioned above. The fix @smurawski suggested didn't work for all our tests (but I changed it for the ones it worked for) so I went ahead and made a work item to investigate this further so it won't block this release

**Comment by panchagnula** [comment]
> @panchagnula I made the change to testing validation you mentioned above. The fix @smurawski suggested didn't work for all our tests (but I changed it for the ones it worked for) so I went ahead and made a work item to investigate this further so it won't block this release

Thanks a lot @StrawnSC , good to see the Diff show more test cases are not live mode anymore. For UP , we need to figure out a better way to make these not live. We can work with core-cli as well during our investigation. These might be tests we need more eyes, so we will still go our manual test pass route.

**Review by panchagnula — APPROVED** [review]
Approved.

**Comment by zhoxing-ms** [comment]
@StrawnSC Could you please address those CI issues?

File: src/containerapp/azext_containerapp/_params.py
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/_params.py:339]: I want to confirm that deleting the default location from resource group will cause a breaking change to users?

Comment (StrawnSC) [reply src/containerapp/azext_containerapp/_params.py:339]: Yes, this is not a breaking change. The API accepts a `null` value for location, and the default location that was being chosen before was causing errors for users

Comment (panchagnula) [reply src/containerapp/azext_containerapp/_params.py:339]: Correct - RG supports a lot more locations than ACA does @zhoxing-ms , hence this change. Thanks for calling this out & checking.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
